[BUG] common/i_crash_posix.cpp:34:10: fatal error: execinfo.h: No such file or directory

Attempting to build Odamex 0.9.5 against musl results in the following failure:

```
FAILED: client/CMakeFiles/odamex.dir/__/common/i_crash_posix.cpp.o 
/usr/bin/x86_64-gentoo-linux-musl-g++ -DCLIENT_APP -DINSTALL_BINDIR=\"bin\" -DINSTALL_DATADIR=\"share\" -DINSTALL_PREFIX=\"/usr\" -DUNIX -I/var/tmp/portage/games-engines/odamex-0.9.5/work/odamex-src-0.9.5/client/sdl -I/var/tmp/portage/games-engines/odamex-0.9.5/work/odamex-src-0.9.5/client/src -I/var/tmp/portage/games-engines/odamex-0.9.5/work/odamex-src-0.9.5/client/../libraries/textscreen -I/var/tmp/portage/games-engines/odamex-0.9.5/work/odamex-src-0.9.5/common/. -I/var/tmp/portage/games-engines/odamex-0.9.5/work/odamex-0.9.5_build/common -isystem /usr/include/SDL2 -isystem /var/tmp/portage/games-engines/odamex-0.9.5/work/odamex-0.9.5_build/libraries/local/include  -DNDEBUG -O2 -pipe -march=x86-64 -frecord-gcc-switches -fno-diagnostics-color -fmessage-length=0 -Wall -Wextra -fdiagnostics-color=always -Wduplicated-cond -Wduplicated-branches -Wrestrict -Wnull-dereference -Wuseless-cast -Wformat=2 -Wno-unused-parameter -std=gnu++98 -MD -MT client/CMakeFiles/odamex.dir/__/common/i_crash_posix.cpp.o -MF client/CMakeFiles/odamex.dir/__/common/i_crash_posix.cpp.o.d -o client/CMakeFiles/odamex.dir/__/common/i_crash_posix.cpp.o -c /var/tmp/portage/games-engines/odamex-0.9.5/work/odamex-src-0.9.5/common/i_crash_posix.cpp
/var/tmp/portage/games-engines/odamex-0.9.5/work/odamex-src-0.9.5/common/i_crash_posix.cpp:34:10: fatal error: execinfo.h: No such file or directory
   34 | #include <execinfo.h>
      |          ^~~~~~~~~~~~
compilation terminated.
```

It looks like this sort of issue has come up in other projects as well: https://github.com/tlx/tlx/issues/36

Oh sorry, I see what you mean. I decided to test this on Alpine Linux. I installed the `libexecinfo` and `libexecinfo-dev` packages using the apk package manager. Although this provided the missing `execinfo.h` and I was able to continue, I ended up hitting a new error about an undefined reference to `backtrace`. I suspect libexecinfo is not completely compatible with glibc execinfo.

Perhaps an alternative solution would be to make this functionality optional in Odamex, then users without these functions can disable it before building.

--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -6,3 +6,12 @@ configure_file(git_describe.h.in "${CMAKE_CURRENT_BINARY_DIR}/git_describe.h")
 add_library(odamex-common INTERFACE)
 target_sources(odamex-common INTERFACE ${COMMON_SOURCES} ${COMMON_HEADERS})
 target_include_directories(odamex-common INTERFACE . ${CMAKE_CURRENT_BINARY_DIR})
+
+if(UNIX)
+  include(CheckSymbolExists)
+  check_symbol_exists(backtrace "execinfo.h" HAVE_BACKTRACE)
+
+  if(HAVE_BACKTRACE)
+    target_compile_definitions(odamex-common INTERFACE HAVE_BACKTRACE)
+  endif()
+endif()

--- a/common/i_crash_noop.cpp
+++ b/common/i_crash_noop.cpp
@@ -23,7 +23,7 @@
 
 
 #if defined _WIN32 && !defined _XBOX && defined _MSC_VER && !defined _DEBUG
-#elif defined UNIX && !defined GEKKO
+#elif defined UNIX && defined HAVE_BACKTRACE && !defined GEKKO
 #else
 
 #include "odamex.h"

--- a/common/i_crash_posix.cpp
+++ b/common/i_crash_posix.cpp
@@ -22,7 +22,7 @@
 //-----------------------------------------------------------------------------
 
 
-#if defined UNIX && !defined GCONSOLE
+#if defined UNIX && defined HAVE_BACKTRACE && !defined GCONSOLE
 
 #include "odamex.h"
 

